src/v/archival/ntp_archiver_service.cc

```diff
@@ -1478,12 +1478,30 @@ ntp_archiver::schedule_single_upload(const upload_context& upload_ctx) {
       upload.sources.empty()
       && upload_ctx.upload_kind == segment_upload_kind::compacted
       && model::offset{} != upload.final_offset) {
-        vlog(
-          _rtclog.warn,
-          "Upload skipped for range: {}-{} because these offsets lie inside "
-          "batches or the reupload is not smaller than the current segment",
+        const auto log_msg = fmt::format(
+          "Upload skipped for range: {}-{}, reason: {}",
           upload.starting_offset,
-          upload.final_offset);
+          upload.final_offset,
+          upload.status);
+
+        switch (upload.status) {
+        case upload_candidate::creation_status::success:
+            // we should never end up here
+            break;
+        case upload_candidate::creation_status::err_offset_inside_batch:
+            vlog(_rtclog.warn, "{}", log_msg);
+            break;
+        case upload_candidate::creation_status::err_size_unchanged:
+        case upload_candidate::creation_status::err_no_segments:
+        case upload_candidate::creation_status::err_head_seek:
+        case upload_candidate::creation_status::err_tail_seek:
+        case upload_candidate::creation_status::err_file_range:
+        case upload_candidate::creation_status::err_no_segment:
+        case upload_candidate::creation_status::err_no_ntp_config:
+        case upload_candidate::creation_status::err_no_content:
```

Comment:
using `[[fallthrough]];` here caused clangd to complain of identical statements

Reply:
bummer, a bug in clangd?

Reply:
possibly, I also see the same warning in `remote::delete_objects_max_keys` etc now. I added `[[fallthrough]]` anyway because it looks like it should be added here.